### PR TITLE
Canonicalize swift header files headers and footers

### PR DESCRIPTION
**Pull request by danra**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

- Added missing ifdef guard in PointerIntEnum header
- Consistent naming convention for ifdef guards
- Consistent 'end namespace swift'
- Consistent single EOL at end of header files

### CODE REVIEW

File: include/swift/Basic/DemangleWrappers.h [hunk 1]
```diff
@@ -57,5 +57,5 @@ demangleTypeAsString(StringRef MangledTypeName,
 } // end namespace demangle_wrappers
 } // end namespace swift
 
-#endif // LLVM_SWIFT_BASIC_DEMANGLE_WRAPPERS_H
+#endif // SWIFT_BASIC_DEMANGLE_WRAPPERS_H
```
Comment (gottesmm) [inline include/swift/Basic/DemangleWrappers.h]: I am not sure if you need a '_' in between Demangle and wrappers. I would look in swift/llvm to see what people do. IIRC they don't do that.

**Comment by tkremenek** [comment]
@swift-ci please test

File: include/swift/Basic/DemangleWrappers.h (hunk 1 above)
Comment (danra) [inline include/swift/Basic/DemangleWrappers.h]: Fixed across all files.

**Comment by danra** [comment]
@tkremenek Could you please merge this?

**Pull request merged**